src/pages/JourneyPage/JourneyPage.ts

```diff
@@ -27,31 +27,53 @@ class JourneyPage extends Base {
 
   isAuthorized : boolean;
 
-  constructor(parent : HTMLElement) {
-
+  constructor(parent, pathParams, isAuthorized) {
     super(parent);
     this.IDs = [];
     this.isOwn = false;
-    this.isAuthorized = arguments[2];
-
-    const pathParams = arguments[1];
+    this.isAuthorized = isAuthorized;
+   
     const journeyType = pathParams[1];
-
+   
     if (pathParams.length > 2) {
       router.go('404');
     }
+   
+    this.isEdit = journeyType === 'edit' || pathParams[0] === 'new';
+    this.type = this.isEdit ? journeyType : 'view';
+    this.tripID = pathParams[0];
 
-    if (journeyType === 'edit' || pathParams[0] === 'new') {
-      this.type = journeyType === 'edit' ? this.type = 'edit' : this.type = 'new';
-      this.tripID = pathParams[0];
-      this.isEdit = true;
-    } else {
-      this.isEdit = false;
-      this.tripID = pathParams[0];
-      this.type = 'view';
+    console.log(journeyType, this.type, this.isEdit, pathParams);
+
+    if (!this.type) {
+      this.type = 'new';
     }
   }
 
+  deleteButtonAdd(id : string) {
```

Comment:
Как будто бы так лучше будет)
```suggestion
  renderDeletePlaceButton(id : string) {
```